src/shared/injector.js

```diff
@@ -64,6 +64,15 @@ export class Injector {
       return this._instances.get(name);
     }
 
+    if (name.endsWith('Factory')) {
+      const [actualName] = name.split('Factory');
+      const instance = () => this.get(actualName);
+
+      this._instances.set(name, instance);
+
+      return instance;
+    }
```

Comment:
This extends the injector so that we can inject services suffixed with `Factory` in a way that not the actual service is returned, but a callback which resolves it at runtime instead.

This allows to address circular dependencies (which I faced when I tried to inject the `FrameSyncService` into the `ToestMessengerService`).

Reply:
I like option 1.

Reply:
I can confirm that inverting the `frameSync`<->`toastMessenger` dependency via events does fix the circular dependency in a very clean way :)